### PR TITLE
feat(workflows): Workflow DX

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
**What**
Provide another DX for the workflow usage, this pr comes with a set of utils as listed bellow

|   |   | 
|---|---|
|  createWorkflow  |  create a new workflow and compose the event flow |   
|  createStep  | create a step and get a step function to compose the workflow  | 
|  transform | transform the data and use the result to provide it to another step  |
|  hook | provides hook into the flow of event and pass use the result for other transformation  |  
|  parallelize | provides an util to parallelize multiple steps  |  
| StepResponse | Class that bundle a step output and compensate input data as the step response |

### CODE REVIEW

**Comment by adrien2p** [comment]
@olivermrbl @srindom you can start reviewing this one if you want

File: packages/workflows/src/utils/composer/create-step.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            const executionContext: StepExecutionContext = {
+              container: transactionContext.container,
+              metadata: transactionContext.metadata,
+              context: transactionContext.context,
+            }
+
+            const invokeResult =
+              transactionContext.invoke[stepName].output?.compensateInput
+
+            const args = [executionContext, invokeResult]
```
Comment (olivermrbl) [inline packages/workflows/src/utils/composer/create-step.ts]: todo: Should be switched around

**Review by olivermrbl — APPROVED** [review]
🚢 🚢 🚢 🚢 🚢 🚢

**Pull request merged**